Userland/Libraries/LibJS/Parser.cpp

```diff
@@ -1507,6 +1507,15 @@ NonnullRefPtr<Expression> Parser::parse_secondary_expression(NonnullRefPtr<Expre
         return parse_assignment_expression(AssignmentOp::NullishAssignment, move(lhs), min_precedence, associativity);
     case TokenType::QuestionMark:
         return parse_conditional_expression(move(lhs));
+    case TokenType::QuestionMarkPeriod:
+        // FIXME: This should allow `(new Foo)?.bar', but as our parser strips parenthesis,
```

Comment:
Yeah these kinds of things are tricky in our parser. If you can think of a good way please let me know because there are more problems like this.